pkg/controller/reconciler/application_reconciler.go

```diff
@@ -0,0 +1,66 @@
+package reconciler
```

Comment:
Have you considered putting this reconciler in a separate directory? E.g. `pkg/controller/reconciler/application/`, next to `pkg/controller/reconciler/cluster/` for the others?

Reply:
Yeah, but couldn't really make up my mind. The certificate reconciler is for example used by both the cluster and applications. Meaning we'd need a `common`, `cluster` and an `application`.

Do chime in if you have any thoughts

Reply:
Right, I knew there was going to be some issues.

Slightly related: In convert.go we have both `CreateResourceDependencyTree` and `CreateApplicationResourceDependencyTree`, so we don't have any clear separation there either. So all together I think we can leave it as it is, it's simple enough.

Reply:
Also a good point. 

I kinda settled on waiting until we got one more case for creating dependency trees / reconcilers to see if there was an obvious pattern